src/instana/instrumentation/pika.py

```diff
@@ -2,166 +2,219 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2021
 
-
-
+from typing import TYPE_CHECKING, Any, Callable, Dict, Tuple
 import wrapt
-import opentracing
 import types
 
-from ..log import logger
-from ..singletons import tracer
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.propagators.format import Format
+from instana.singletons import tracer
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 try:
     import pika
 
+    if TYPE_CHECKING:
+        import pika.adapters.blocking_connection
+        import pika.channel
 
-    def _extract_broker_tags(span, conn):
-        span.set_tag("address", "%s:%d" % (conn.params.host, conn.params.port))
+    def _extract_broker_tags(span: "InstanaSpan", conn) -> None:
+        span.set_attribute("address", f"{conn.params.host}:{conn.params.port}")
 
-
-    def _extract_publisher_tags(span, conn, exchange, routing_key):
+    def _extract_publisher_tags(
+        span: "InstanaSpan", conn, exchange, routing_key
+    ) -> None:
         _extract_broker_tags(span, conn)
 
-        span.set_tag("sort", "publish")
-        span.set_tag("key", routing_key)
-        span.set_tag("exchange", exchange)
-
+        span.set_attribute("sort", "publish")
+        span.set_attribute("key", routing_key)
+        span.set_attribute("exchange", exchange)
 
-    def _extract_consumer_tags(span, conn, queue):
+    def _extract_consumer_tags(span: "InstanaSpan", conn, queue) -> None:
```

Comment:
Typing annotation is missing for conn and queue

Reply:
Done